### PR TITLE
add docs for querying Redis state store

### PR DESCRIPTION
**Pull request by dmitsh**
Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [x] [Read the contribution guide](https://docs.dapr.io/contributing/contributing-docs/)
- [ ] Commands include options for Linux, MacOS, and Windows within codetabs
- [ ] New file and folder names are globally unique
- [x] Page references use shortcodes instead of markdown or URL links
- [ ] Images use HTML style and have alternative text
- [ ] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

add docs for querying Redis state store

## Issue reference

#2187

### CODE REVIEW

File: daprdocs/content/en/reference/api/state_api.md [hunk 1]
```diff
@@ -289,10 +289,10 @@ None.
 ### Example
 
 ```shell
-curl -X "DELETE" http://localhost:3500/v1.0/state/starwars/planet -H "If-Match: xxxxxxx"
+curl -X DELETE http://localhost:3500/v1.0/state/starwars/planet -H "If-Match: xxxxxxx"
 ```
 
-## Query state
+## Query State
```
Comment (msfussell) [inline daprdocs/content/en/reference/api/state_api.md]: Headings are lower case for following words
```suggestion
## Query state
```

File: daprdocs/content/en/reference/components-reference/supported-state-stores/setup-redis.md [hunk 2]
```diff
@@ -81,7 +83,8 @@ If you wish to use Redis as an actor store, append the following to the yaml.
 | idleCheckFrequency        | N        | Frequency of idle checks made by idle connections reaper. Default is `"1m"`. `"-1"` disables idle connections reaper. | `"-1"`
 | idleTimeout        | N        | Amount of time after which the client closes idle connections. Should be less than server's timeout. Default is `"5m"`. `"-1"` disables idle timeout check. | `"10m"`
 | actorStateStore    | N         | Consider this state store for actors. Defaults to `"false"` | `"true"`, `"false"`
-| ttlInSeconds       | N         | Allows specifying a default Time-to-live (TTL) in seconds that will be applied to every state store request unless TTL is explicitly defined via the [request metadata]({{< ref "state-store-ttl.md" >}}). | `600` 
+| ttlInSeconds       | N         | Allows specifying a default Time-to-live (TTL) in seconds that will be applied to every state store request unless TTL is explicitly defined via the [request metadata]({{< ref "state-store-ttl.md" >}}). | `600`
+| queryIndexes       | N         | Query indexing schemas for querying JSON objects |
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-state-stores/setup-redis.md]: Can you provide an example here in the last column

File: daprdocs/content/en/reference/components-reference/supported-state-stores/setup-redis.md [hunk 3]
```diff
@@ -147,6 +150,108 @@ We can use [Helm](https://helm.sh/) to quickly create a Redis instance in our Ku
 
 {{< /tabs >}}
 
+### Querying JSON objects
+
+Redis state store supports querying JSON objects. To enable this feature, the following steps are required:
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-state-stores/setup-redis.md]:
```suggestion
The Redis state store supports querying of JSON objects. To enable this feature, the following steps are required:
```

File: daprdocs/content/en/reference/components-reference/supported-state-stores/setup-redis.md [hunk 4]
```diff
@@ -147,6 +150,108 @@ We can use [Helm](https://helm.sh/) to quickly create a Redis instance in our Ku
...
+1. Load [redisearch](https://oss.redis.com/redisearch/) and [rejson](https://oss.redis.com/redisjson/) modules when deploying redis service.
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-state-stores/setup-redis.md]:
```suggestion
1. The Redis store must support Redis modules and specifically both Redisearch and RedisJson. If you are deploying and running Redis then load [redisearch](https://oss.redis.com/redisearch/) and [redisjson](https://oss.redis.com/redisjson/) modules when deploying the Redis service.
``

**Comment by msfussell** [comment]
@dmitsh - It is important here to help people understand how they can set up Redis correctly for querying with some common Redis stores and hence update https://docs.dapr.io/reference/components-reference/supported-state-stores/setup-redis/#setup-redis section

- For Selfhosted what Redis CLI commands can you run to deploy these the Redis modules yourself after Dapr init. Being able to easily do this locally is a great example. Please add the commands here.
-  Same for K8s, Redis CLI commands
- Under Azure can you put a note" Note: Azure currently [does not support the RedisJson](https://docs.microsoft.com/en-us/answers/questions/333933/does-azure-cache-for-redis-support-any-of-the-redi.html) module and cannot be used with query.
-  Under AWS can you put - "Note: For query support you need to enable RediSearch and RedisJson."
-  Under GCP can you put a note" Note: Memory Store does not modules and cannot be used with query."
- Can you also add new Tab for Redis Enterprise Cloud with this link https://docs.redis.com/latest/rc/ as another option. 
- Can you also add new Tab Alibaba Cloud with this link https://www.alibabacloud.com/product/apsaradb-for-redis as another option

**Review by msfussell — CHANGES_REQUESTED** [review]

**Comment by msfussell** [comment]
I would like to make sure that this and the query API topic https://docs.dapr.io/developing-applications/building-blocks/state-management/howto-state-query-api/ article are better aligned. Could you show the Redis queries using the employees dataset (have a link to this in this Redis component topic) and using query from Example 1, In other words, this

curl -X POST http://localhost:3500/v1.0/state/statestore/query?metadata.contentType=application/json&metadata.queryIndexName=orgIndx -d @query-api-examples/query1.json -H "Content-Type: application/json

Can you also add a link in the "Related Links" section to this Redis Component Query section? 
[Query Redis State Store](link)

**Comment by msfussell** [comment]
These changes look great! Would still like to address what Redis service can be used due to the Redis Modules dependency.
These ones.

> * Under Azure can you put a note" Note: Azure currently [does not support the RedisJson](https://docs.microsoft.com/en-us/answers/questions/333933/does-azure-cache-for-redis-support-any-of-the-redi.html) module and cannot be used with query.
> * Under AWS can you put - "Note: For query support you need to enable RediSearch and RedisJson."
> * Under GCP can you put a note" Note: Memory Store does not modules and cannot be used with query."
> * Can you also add new Tab for Redis Enterprise Cloud with this link https://docs.redis.com/latest/rc/ as another option.
> * Can you also add new Tab Alibaba Cloud with this link https://www.alibabacloud.com/product/apsaradb-for-redis as another option

**Comment by dmitsh** [comment]
> These changes look great! Would still like to address what Redis service can be used due to the Redis Modules dependency. These ones.
> 
> > * Under Azure can you put a note" Note: Azure currently [does not support the RedisJson](https://docs.microsoft.com/en-us/answers/questions/333933/does-azure-cache-for-redis-support-any-of-the-redi.html) module and cannot be used with query.
> > * Under AWS can you put - "Note: For query support you need to enable RediSearch and RedisJson."
> > * Under GCP can you put a note" Note: Memory Store does not modules and cannot be used with query."
> > * Can you also add new Tab for Redis Enterprise Cloud with this link https://docs.redis.com/latest/rc/ as another option.
> > * Can you also add new Tab Alibaba Cloud with this link https://www.alibabacloud.com/product/apsaradb-for-redis as another option

I would prefer not to mention that in the doc because:
 - things can change, and this information may become inaccurate
 - you can still deploy Redis with modules in Azure, for example. I did that in e2e tests with AKS

**Comment by msfussell** [comment]
Let's at least then be inclusive of other Redis hosted services and add tabs to reference Redis Cloud and Alibaba Cloud

> * Can you also add new Tab for Redis Enterprise Cloud with this link https://docs.redis.com/latest/rc/ 
> * Can you also add new Tab Alibaba Cloud with this link https://www.alibabacloud.com/product/apsaradb-for-redis

**Review by msfussell — APPROVED** [review]
Fantastic!

**Pull request merged**